### DISCUSSION TITLE
장바구니 협업 미션 1단계 상호 코드 리뷰 | 2022.06.03

### DISCUSSION THREAD

**Discussion by wmakerjun** [woowacourse/retrospective#17] [General]
# 상호 코드 리뷰 회고
- 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.
- 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

페어별로 위 2가지 질문들에 대해 이야기 나누고 1개의 코멘트에 내용을 정리해서 공유해주세요.

**Comment by juunzzi** [comment]
1. 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요. ☺️☺️

준찌: 확실히 같은 미션을 구현한 다른 코드를 보니깐 생각이 많아져서 좋았어요 !!! 나는 어떻게 했지? 요렇게 하면 어떤게 내 방식보다 좋을까를 조금 생각해보았다? -> 메타인지에 도움이 된 것 같아요

해리: 준찌와 비슷한 맥락! 자신이 모자르다고 생각했던 부분을 다른 사람을 통해 비춰볼 수 있었다. 내 방식과 다른 사람의 방식을 비교해보고 얻은 점들이 있었다.

2. 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요? ☺️☺️

준찌: 

요청하는 사람 입장 - 미리 구현의도를 (남이 궁금해할수있는) 적어둔다면? 더 효과적으로 코드리뷰를 할 수 있을 것 같다. 불필요한 의사소통이 사라져서 더 좋을 것 같습니다! 

리뷰를 달아주는 사람 입장 - 틀렸다고 생각하고 리뷰해주기보단 다르다고 생각하고 리뷰해주는 것이 좋을 것 같다. 내 방식이 정답일 수 없으니까 ! 감정적으로 안좋아지거나 사고가 닫히게 되어 좋지 않은 것 같습니다.

해리: 

요청하는 사람 입장 - 어떤 점을 중점으로 봐줬으면 좋겠는지 작성해준다면 ? 더 좋을 것 같다. 핵심 코드들을 미리 적어준다면 리뷰하는 입장에서 시간을 아낄 수 있을 것 같다. 그 부분을 중점적으로 보게된다면 양질의 리뷰가 나올지도 ?

리뷰를 달아주는 사람 입장 - 리뷰를 많이 달자! 한번 쯤 더 생각하게 되어서 일단은 리뷰의 양이 많았으면 좋겠다고 생각합니다. 고민을 하게끔 하는 리뷰가 많다면 좋겠죠 ?

**Comment by moonheekim0118** [comment]
# 👨‍🏫 코이 & 😎호프

## ✅ 느낀점
### 코이
- 생각보다 어려웠다.
- 많이 배우기도 했다.

### 호프
- 코드리뷰를 통해 간접적으로나마 다른 사람의 생각을 엿볼 수 있어서 즐거웠다.
- 시간이 생각보다 오래 걸린다.


## ✅ 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

### 코이
- 리뷰를 받는 입장 : 무조건 수정하기보다 생각을 하는게 좋을 것 같고, 남긴 리뷰에 대해서 좋은 점은 좋았다고 얘기하는게 좋을 것 같다.
- 리뷰를 하는 입장 : 반영 했으면 하는 사항과 반영 해야하는 사항을 구분해야 할 것 같다. 

### 호프
- 리뷰를 받는 입장 : 생각보다 리뷰어의 사견이 많이 들어갈 수 있으므로, 무조건 수용하기 보다 내 생각을 거치고 수정해야 할 것 같다. 그리고 너무 깊은 추상화는 리뷰어에게 고통이 될 것 같다..
- 리뷰를 하는 입장:  반영 했으면 하는 사항과 반영 해야하는 사항을 구분해야 할 것 같다.

**Comment by hwangstar156** [comment]
# 도밍!🐠🔪
![image](https://user-images.githubusercontent.com/85891751/171780566-79d49df3-1b1b-48d3-996d-54059b4a3d63.png)


## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

스밍: 생각보다 리뷰하는것이 어려웠었다! 남의 코드를 이해하고 어떤 목적으로 사용했는지 몰라서 리뷰하기가 조심스러웠다. 어디를 변경하고 어디를 변경안한지 file changes가 많아지니 확인하기 어려웠다.
도링: 생각보다 빠르게 남의 코드를 이해하는게 쉽지 않았다. 1줄이상 상세하게 리뷰를 남기는 일이 어려웠다.

## 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

스밍: 요청하는 사람은 pr에 자신이 다 못한것이나 스타일 가이드등을 미리 리뷰어에게 알려주어 이미 알고있는것에 대해서 리뷰를 중복해서 안받게 하는것이 좋을것 같다. 자신이 봐줬으면 하는 부분을 pr에 적으면 좋을것 같다. 리뷰어의 입장에서는 이런식으로 해라 보다는 리뷰 받는사람의 의견을 물어보는듯한 리뷰가 좋을것 같다.

도링: 리뷰를 받는 입장에서 어떤 점을 받고 싶은지, 무엇을 생각하며 작성한 코드인지를 pr에 적는 것이 좋을 것 같다. 리뷰어의 입장에서는 리뷰이가 작성한 코드를 이렇게 바꿔주세요가 아니라 다시 생각해볼 수 있는 코멘트를 남기는 것이 좋다

**Reply by prefer2** [reply]
사진 너무 크자너ㅋㅋㅋ

**Reply by rladpwl0512** [reply]
도리 얼굴 너무 커 ㅜ

**Reply by hwangstar156** [reply]
도리도 크잖어

**Reply by compy-ryu** [reply]
사진 너무 커서 놀랐습니다 ㅋㅋㅋㅋ

**Reply by moonheekim0118** [reply]
깜짝이야

**Reply by onschan** [reply]
사진 사이즈 줄여주세요

**Reply by KangYunHo1221** [reply]
화질좋네요

**Comment by yunjin-kim** [comment]
# 샐결👍

리뷰
<img width="53" alt="스크린샷 2022-05-30 오후 5 24 20" src="https://user-images.githubusercontent.com/79692272/171780707-8287ab25-5994-4a80-9817-6c10fe331623.png">
<img width="53" alt="스크린샷 2022-05-30 오후 5 24 20" src="https://user-images.githubusercontent.com/79692272/171780707-8287ab25-5994-4a80-9817-6c10fe331623.png">
<img width="53" alt="스크린샷 2022-05-30 오후 5 24 20" src="https://user-images.githubusercontent.com/79692272/171780707-8287ab25-5994-4a80-9817-6c10fe331623.png">
<img width="53" alt="스크린샷 2022-05-30 오후 5 24 20" src="https://user-images.githubusercontent.com/79692272/171780707-8287ab25-5994-4a80-9817-6c10fe331623.png">

1. 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

  - 리뷰어의 심정을 많이 알 수 있었던 시간이었습니다.
  - 우연히 같은 base인 코드를 가진 상태로 리뷰를 하게 되어서 그나마 코드 이해가 쉬웠다.

3. 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

  - 요청하는 사람 
     - 읽기 쉬운 코드를 작성하는 것이 좋다. 과도한 추상화는 리뷰어의 고통은 배가 된다.  
     - 궁금한 코드가 있을 땐 해당 코드를 복사해서 pr 메세지에 질문과 함께 명시하면 좋을 것 같다.

  - 리뷰를 달아주는 사람
    - 리뷰 코멘트를 사소하더라도 많이 달라주었으면 좋겠다.
    - 과도한 리소스를 남기는 것은 좋지 못하다. 생각할 수 있도록 유도하는 것이 중요하다!

**Reply by liswktjs** [reply]
리뷰 해줘잉~

**Reply by moonheekim0118** [reply]
'해줘'

**Reply by usageness** [reply]
'줘'

**Reply by nan-noo** [reply]
'ㅝ'

**Comment by rladpwl0512** [comment]
# 콤피🦖 & 위니💛  
![images_compy_post_7d441ee1-5d63-45d3-b599-7e2f92647162_역시-리뷰어님이야](https://user-images.githubusercontent.com/52344833/171780444-4fcedf7b-aae5-462d-9dec-81244b5acefe.png)
## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.
- 리뷰어님들의 고충을 알 수 있었다. (리뷰어님들... 죄송해요 😅) 
- 다른 사람의 코드를 판단하는 것이 힘들다. (이 코드를 작성한 의도가 무엇인지 알 수 없어서, 리뷰를 섣불리 남기기 힘들었다)
- 서로 버그를 발견해줄 수 있었다. 
- 코드에서 배울 점을 발견할 수 있었다. 

## 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?
### 요청하는 사람 
- 가독성 좋은 코드, 명확한 네이밍을 작성한다 
- 애매한 부분이나 특별한 의도가 있었던 부분이 있다면, 댓글로 의도를 설명해주는 것이 좋을 것 같다 

### 리뷰를 달아주는 사람 
- 바로 리뷰를 남기기 보다는, 의도를 한번 파악해보는 과정이 있으면 좋을 것 같다 (의도가 이해되지 않는 경우에는, 댓글로 물어본다)  
- 내 방식을 강요하기보다는, 내가 남긴 리뷰에 대해 어떻게 생각하는지 물어보는 과정이 있으면 서로 배울 수 있을 것 같다.

**Reply by KangYunHo1221** [reply]
신성한 리뷰시간에 짤만드느라 시간을 허비하다니 콤피..!!!

**Reply by wonsss** [reply]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ넘 웃기네요

**Reply by usageness** [reply]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ짤 뭐야

**Comment by lokba** [comment]
# 클릭하면 근성장, 내일 일어나면 근뿜뿜 클릭 안하면 근손실 100%
## 😎😎😎😎😎😎록스타😎😎😎😎😎😎


- From 록바 To 온스타
- 다른 방식으로 폼을 구현해서, 다양한 맛을 맛 볼수 있어서 좋았습니다. 코드의 일관성이 잘 되어있어 읽기 편했습니다. 온스타의 코드를 보면서 제 코드의 부족한 부분을 알았고, 수정할 계획입니다. 
- 커밋 메시지 또는, 기본적으로 신경써야 하는 부분을 필수적으로 할 필요가 있는 것 같다. 



- From 온스타 To 록바
- 동일한 기능 구현에서, 다른 방식으로 접근한 것을 보면서 많이 느꼈고 배웠다. 도움이 된 것 같다. 리팩토링을 추후에 완료한 후, 코드를 보고 싶다. 
- 리뷰를 달때, 정답을 제시하는 것이 아니라 다양한 생각을 경험할 수 있는 방향으로 해주는 것이 좋은 것 같다. 


<p align="center">
<img src="http://openimage.interpark.com/goods_image_big/6/2/0/2/7706076202_l.jpg"  width="50%"/>
온스타
<img src="https://ae01.alicdn.com/kf/Ha9360014eb644ecbb9173b73080aea09e.jpg?width=800&height=800&hash=1600"  width="50%"/>
록바
</p>

**Reply by hwangstar156** [reply]
으악 이거 뭐야

**Reply by hwangstar156** [reply]
![image](https://user-images.githubusercontent.com/85891751/171780434-04024408-7bd6-41ac-93cb-ea3d42d28e14.png)

**Reply by yunjin-kim** [reply]
상체에 비해 하체가 부실하네요~

**Reply by lokba** [reply]
피지크 입니다.

**Reply by compy-ryu** [reply]
아니 선생님들...

**Reply by KangYunHo1221** [reply]
일본산 피규어 인거 같아서 비추합니다.

**Reply by kwannee** [reply]
너무 징그러워서 비추 드렸어요~

**Reply by hwangstar156** [reply]
저도 광희씨 비추요

**Comment by DomMorello** [comment]
## 돔하디 vs 시지프
### 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.
`돔하디`: 다른 사람의 코드를 보면서 구조를 파악하는 것이 생각보다 어렵다. 코드리뷰를 제대로 하려면 시간이 많이 필요하다.
`시지프`: 설명없이 코드만 보는게 힘들다. 다음에 리뷰어를 위하여 의도 설명과 함께 코드를 제출할 것 같다.

### 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?
`돔하디`: 리뷰를 달아주는 사람은 충분한 시간을 갖고 코드를 이해해야 한다. 시간이 없으면 얕은 수준의 코드리뷰만 된다. 요청하는 사람은 기능 문제를 면밀하게 파악한 후에 안 되는 사항 및 미흡한 사항을 정리해서 알려주면 좋을 것 같다. 
`시지프`: 요청하는 사람은 코드의 의도를 설명하자. 파일구조를 생각하고 짜야되고 범용적인 방법을 사용하는 것이 좋을 것 같다. 그래야 리뷰어가 파일 구조를 쉽게 파악할 수 있다. 

### 의문점
코드리뷰하는 입장에서 커밋단위를 봐야 할 일이 있을까?

**Reply by DomMorello** [reply]
저희는 의문점도 있죠?

**Comment by jswith** [comment]
# 태태x민초

## 직접 다른 사람의 1단계 PR 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

### 태태

- 내가 아는 게 부족해서 뭘 써야할 지 생각하기 어려웠다. 주로 내 생각을 썼는데, 경험이 더 많을수록 좋은 리뷰를 남길 것 같다.

### 민초

- 무수한 File Changes에 압도되었어요. 리뷰어분들 존경스럽네요.
- 코드 리뷰가 생각보다 쉽지 않네요. 일일이 읽어보는게 쉽지 않은 것 같아요. ㅠㅠ

## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

### 태태

- 요청하는 사람이 어떤 생각을 했고, 어떤 식으로 구현했는지 작성하면 리뷰어가 편할 것 같다. 리뷰어는 리뷰이가 말한 내용을 정확히 파악하는 게 좋을 듯! 엉뚱한 답을 안 하도록 노력해야할 것 같다.

### 민초

- 리뷰를 달아주는 사람 : 리뷰이가 어떤 생각으로 코드를 작성했는지 이해한 후 코드리뷰
- 리뷰를 요청하는 사람 : 리뷰의 범위를 좁히기 위해 코드를 어떤 생각으로 구현했는지 리뷰어에게 설명하고, 시행착오, 해결과정, 궁금한 점 등을 최대한 남길 것

**Comment by dayelop** [comment]
# ✨ ❎ 🐯

## 느낀점

- 어디서부터 어디까지 리뷰해야 할지 막막했다
- 옆에서 바로 물어볼 수 있어서 코드의 의도를 파악하면서 리뷰하기에 좋았다
- 말하고 싶은 내용을 어떻게 코멘트로 남길지 고민스러웠다
- ✨ 코드 구경하기 재밌다

## 개선점

### 리뷰이

- 폴더 구조를 정리하자
- 코드 가독성을 높이자
- 코드의 일관성을 지키면 리뷰하는 데에 힘을 덜 들이겠다~!
- 리뷰어에게 중점적으로 리뷰 받기를 원하는 부분을 알린다면 궁금한 부분에 대해서 많이 리뷰를 받을 수 있겠다

### 리뷰어

- 코드 레벨의 취향이 있더라도 상대방의 의도를 먼저 물어보자
- 코드를 입맛대로 바꾸려고 하지 말자
- 리뷰어의 입장이니까 아무래도 내용이 강압적일 수 있으니 말투에 신경 쓰자
- 리뷰어... 빡세겠다

**Reply by uk960214** [reply]
재밌었다 🧐

**Reply by dayelop** [reply]
드디어 만난 블티 ✨🐯

**Comment by jin7969** [comment]
### 😄 안 😄
코드를 이해하는 부분에서 시간이 많이 걸리고 어려움이 있어서 다른 사람의 코드를 읽어보는 연습을 해야할 거 같습니다.
또한, 리뷰를 받는 과정에서 소통이 얼마나 중요한지 깨달았습니다. 다음부터는 무엇을 중심으로 생각하였고, 질문에 대한 내용을 조금더 자세하게 적성을하면
리뷰어 입장에서 수월하게 리뷰를 할 수 있다고 생각하였습니다 ㅎㅎ

### 🥤 코카콜라 🥤

리뷰어 분들께 죄송한 마음이 들었습니다.
그동안 아무 질문 없이! 설명 없이! PR을 날렸는데, 막상 제가 리뷰를 할려고 하니 어디부터 봐야 하는지, 어디가 궁금한 곳인지 알수가 없어서 코드 전체를 처음부터 봐야 하는 수고를 느끼게 되었습니다. 그래서 매우 반성하고 있습니다. ㅠㅠ

**Comment by wonsss** [comment]
# 밧드 x 마르코

## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

- 많은 것을 배울 수 있었다. 생각보다 비슷한 로직도 있었고 다른 로직도 많았는데, 각각을 비교하고 장단점을 논의하면서 더 좋은 코드가 무엇일지 한 발짝 다가갔을지도...(응 맞아)
- 인풋 컴포넌트 같은 경우 많이 고민했는데, 확장성을 어떻게 늘릴 수 있을까 고민했었다. 코드리뷰해보면서 상대의 코드에서 많은 영감을 받게 됐다. 

## 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?
- 요청하는 사람 입장
	- 자기가 무엇을 고민했는지 자세히 적어줘야한다. 자기 코드가 아니기 때문에 리뷰어를 배려해서 관련된 코드에 직접 명시해주면 찾기 쉽다.
	- 요청하는 사람 입장에서는 코드를 왜 이렇게 작성했는지 의도를 최대한 자세히 작성해야 한다. 
- 리뷰를 달아주는 사람
	- 이해가 잘 되지 않더라도 해결책이 아닐 수 있다는 두려움을 가지고 스스로 검열하기 보다 의문가는 점이 있다면 물어보면 좋겠다. 상대가 어떤 목표를 달성하기 위해 코드를 작성했는지 이해하려는 목표를 갖고 리뷰하면 좋을 것 같다.
	- 리뷰를 달아주는 사람 입장에서는, 본인의 리뷰가 틀릴 수 있다고 괜히 두려워하기보다 궁금한 것을 물어본다는 관점으로 최대한 자유롭게 많이 리뷰를 다는 것도 좋을 것 같다.

**Comment by byhhh2** [comment]
## 🙃병민 X 무비🙂

### 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

- 어려운 로직 + 개행 없음 => 이해 불가 
- 리뷰어분들이 코드 리뷰하기 힘들었겠다.. 깃허브에서 리뷰남기는 것이 조금 불편했을 것 같다.
- 윤병인씨 : 양심에 찔렸다. 왜냐? 나도 깔끔하게 못짜는데 리뷰하고 있어서.. 근데 재밌었다. 😆
- 변영화씨 : 나와 같은 주제를 가지고 다른 코드를 읽어보니 특정 부분에서 다른 점이 있는 것을 발견하니 너무 재밌었다. 그치만, 코드가 너무 어려웠다. 그래서 useForm 구체적인 로직이 있는 부분은 버리고, page부터 읽었는데 그 부분은 나쁘지 않았다. (알고보니 난 추상화된 부분만 본 것이다.) 

### 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

#### 요청하는 사람

- 불필요한 주석보다는 함수명을 구체적으로 작성한다. 
- 리뷰를 요청하기 전에 자신의 코드를 먼저 셀프리뷰해본다.

#### 리뷰를 달아주는 사람 

- 리뷰이가 어떤 부분이 부족한 지 판단하여 방향성을 추천해준다. 
- 코드를 꼼꼼하게 보아야한다.
- 사소한 부분이라도 짚어주어야 한다.

**Reply by airman5573** [reply]
진짜 회고 잘했다...

**Reply by usageness** [reply]
인정,,,

**Reply by airman5573** [reply]
소중한 한표 부탁 드립니다!

**Comment by al-bur** [comment]
## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.

### 유세지

짧은 시간내에 유기적으로 연결된 코드를 파악하기가 생각보다 어려웠습니다.
커밋 단위로 코드를 보면 이후에 수정된 부분을 피드백해주기 난감한 경우가 있습니다.

### 앨버

시간이 짧아서 추가된 코드를 보기위해 commit 기반으로 바뀐 파일들을 보면서 리뷰를 했었는데, 지엽적인 코드들을 볼 수는 있었지만 전체적인 구조등은 볼 수 없었다...
전체적인 구조를 보기 위해서는 개발 실력이 뛰어나야 가능한걸까라는 생각이든다..

## 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?

### 유세지

리뷰를 요청하는 측에서 특별히 신경썼던 점이나, 궁금했던 점을 커밋과 함께 남겨주면
그 부분을 놓치지 않고 집중적으로 답변해줄 수 있어서 좋을 것 같습니다.

리뷰를 달아주는 입장에서는 전체적인 구조에 대한 피드백을 해줄지,
지엽적인 (디테일) 부분을 짚어주어야 할지 고민이 있을 것 같습니다.
프로젝트를 살펴보고 리뷰어가 어려워하는 쪽을 짚어주면 좋을 것 같습니다.

### 앨버

요청할 때, 어떤 부분을 중점적으로 봐달라고 요청을 하는 것이 리뷰를 달아주는 사람이 리뷰하기에 선택과 집중을 할 수 있을 것 같다. 
리뷰를 달아주는 사람은 답을 제시해주는 것보다는 질문을 던져서 코드를 작성한 사람에게 코드에 대해 한 번 더 생각을 해보게 해주는 것이 좋을 것 같다.

**Reply by al-bur** [reply]
<논란이 있어 삭제하였습니다>

**Reply by usageness** [reply]
여러분의 소중한 한 표 감사하빈다 👏

**Reply by hwangstar156** [reply]
🥲

**Reply by KangYunHo1221** [reply]
코인인가요 이거?

**Reply by al-bur** [reply]
물의를 일으켜 모든 크루분들께 죄송합니다 😢

**Comment by soyi47** [comment]
# 😎 비녀 & 소피아 😎

## 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀 점
### 비녀
코리스도 했고, 크루들 코드도 종종 보는데 다른 사람들 코드 보는 경험이 있어서 피로도는 덜했지만 여전히 어렵다. 그런데 재미있다. 리뷰를 달면서 꿀팁을 얻을 수 있다는 점이 아주 좋다. 리뷰를 하면서 성장하는 부분이 큰 것 같다.
### 소피아 
남의 코드 보기 참 힘들다. 정신이 맑을 때 봐야 효율적으로 잘 볼 수 있을 것 같다. 코드를 보면서 셀프 반성과 꿀팁을 주워가는 점들, 그리고 전혀 생각 못 해봤던 방식을 얻어가는 부분들이 좋다.

## 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?
### 비녀
리뷰이의 입장
  - 파일 변경에 대한 내용을 같이 공유하는 게 좋겠다.
  - 요구사항 정리된 내용도, 설명서처럼 리뷰어에게 공유하는 게 좋겠다.
  - 코드가 한 파일에 너무 길지 않은 게 좋겠다. 왜 파일 분리가 필요한지 알겠다. 길면 보기 힘들다.

리뷰어의 입장
  - 코드의 구조에 대한 이해를 충분히 한 후에 리뷰를 다는 게 좋겠다. 구조나 흐름에 대한 충분한 이해 없이 달면 코드 스타일이나 환경 설정 위주만 신경쓰게 될 것 같다.

### 소피아
리뷰이의 입장
  - 내가 어떤 의도나 고민을 가지고 코드를 작성했는지, 이번 작업에서 중요하게 생각했던 부분들을 먼저 공유하는 게 중요한 것 같다. 리뷰어가 구조과 고민을 더 잘 이해하고 좋은 리뷰를 달아줄 가능성이 커진다.  
  - 파일 변경, 수정 반영한 커밋 링크를 상세히 공유한다.

리뷰어의 입장
  - 적절한 소통을 위해 노력한다. 리뷰를 진행하면서 잘 이해가 안 되고 의도 불분명하다고 느껴지는 부분들을 리뷰이와 소통하면서, 해소하고 리뷰를 진행하면 양질의 리뷰가 될 수 있을 것 같다.

**Comment by LAH1203** [comment]
## 우디 𝖷 나인 𝖷 하리

- 직접 다른 사람의 1단계 PR에 코드에 리뷰를 하면서 느낀점이 있다면 자유롭게 공유해주세요.
  - 코드 리뷰 너무 힘들다.. 다 잘해보인다
  - 쓴 소리 잘 못 하겠다
  - 좋은 리뷰(피드백)하기가 너무 어려운 것 같다
  - 내 의견을 구체적이고 명확하게 적기가 힘들다
  - 코드 리뷰 스터디는 착한 맛, 코드 리뷰는 매운 맛
- 코드리뷰를 효과적으로 하기 위해서는 요청하는 사람, 리뷰를 달아주는 사람 입장에서 무엇을 신경쓰면 좋을까요?
  - 요청하는 사람
    - 네이밍이 중요한 것 같다. 내용을 보지 않아도 이름으로 무슨 역할을 하는지 판단할 수 있게!
    - 주석도 나쁘지만은 않은 듯 👀
    - 구현 설명이 필요할 것 같다
    - 파일, 코드, 커밋, 네이밍 컨벤션에 대해 적어주면 코드를 이해하기에 좋을 것 같다
    - 리뷰를 받고 싶은 부분을 명확하게 해주면 서로에게 좋을 듯
    - 커밋 단위를 나눠서 알려주는 부분도 있으면 굿
  - 리뷰를 달아주는 사람
    - 리뷰가 보다 구체적이고 명확해야 한다
    - 제안 사항을 제공하는 것이 좋을 듯
    - 칭찬할 것은 칭찬하기 👍
    - 서로 간의 소통(티키타카) 필요
